### ISSUE TITLE
Not possible to debug a Flutter webapp in Chrome not started by DartCode

### ISSUE THREAD

**Issue by maks** [Dart-Code/Dart-Code#4208]
Per https://github.com/flutter/flutter/pull/44263 I am launching a Flutter webapp with only the webserver running, open the url for the webapp in a specific Chrome window (ie. I want to use a Chrome window for a specific Chrome profile which is logged into a specific Google account) and then able to use the VSC debugger to debug the running Flutter webapp.

I'm using this launch config:
```json
{
      "name": "My App Web (no browser launch)",
      "request": "launch",
      "type": "dart",
      "deviceId": "web-server",
      "program": "lib/main_preview.dart",
      "toolArgs": [
        "--start-paused",
        "--dart-define",
        "MYAPP_X_MODE=true",
        "--web-port=42424",
      ]
    },
```

And it all seems to work, hot-restart works and I can set and hit breakpoints, *but* when I hit a breakpoint, in VSC it doesn't jump to the source file location where the breakpoint is set as it normally would. I do still see the variables view correct information and if I manually go to the source where the hit breakpoint is set it shows the way I expect:
![Screenshot from 2022-10-07 11-34-13](https://user-images.githubusercontent.com/71999/194442190-5ae3bc1d-01e0-4e26-89f5-2773571ac6cd.png)

and likewise in the BREAKPOINTS window:
![Screenshot from 2022-10-07 11-33-57](https://user-images.githubusercontent.com/71999/194442259-3959f38f-57b4-43a2-a1de-2d26f9243a53.png)

Also I do note that the CallStack window looks a bit strange with the just all zeros at the end of each frame line:
![Screenshot from 2022-10-07 11-27-02](https://user-images.githubusercontent.com/71999/194441815-be33c9de-fa74-46f8-a5c5-dbef269587e7.png)

**Please complete the following information:**
 - Operating System and version: Ubuntu 22.04
 - VS Code version: 1.71.2
 - Dart extension version: 2.18.1
 - Dart/Flutter SDK version: 3.3.3
 - Target device (if the issue relates to Flutter debugging): Chrome 106.0.5249.91

**Comment by DanTup** [comment]
I can reproduce this. Using the Chrome device is fine, but using Web Server and the Dart Debug extension has the behaviour above.

It looks like when we pause, we call `getScript` to get the script to map the `tokenPos` back to a location. When using the Chrome device, this works:

```
[12:34:13 PM] [VmService] [Info] [Flutter (Chrome)] ==> {"id":"15","jsonrpc":"2.0","method":"getObject","params":{"isolateId":"1","objectId":"4"}}
[12:34:13 PM] [VmService] [Info] [Flutter (Chrome)] <== {"jsonrpc":"2.0","result":{"type":"Script","id":"4","uri":"package:dartcode_flutter_4208/main.dart","library":{"type":"@Library","id":"package:dartcode_flutter_4208/main.dart","name":"package:dartcode_flutter_4208/main.dart","uri":"package:dartcode_flutter_4208/main.dart"},"source":"import 'package:flutter/material.dart';\n\nvoid main() {\n  runApp(const MyApp());\n}\n\nclass MyApp extends StatelessWidget {\n  const MyApp({super.key});\n\n  // This widget is the root of your application.\n  @override\n  Widget build(BuildContext context) {\n    return MaterialApp(\n      title: 'Flutter Demo',\n      theme: ThemeData(\n        // This is the theme of your application.\n        //\n        // Try running your application with \"flutter run\". You'll see the\n        // application has a blue toolbar. Then, without quitting the app, try\n        // changing the primarySwatch below to Colors.green and then invoke\n        // \"hot reload\" (press \"r\" in the console where you ran \"flutter run\",\n        // or simply save your changes to \"hot reload\" in a Flutter IDE).\n        // Notice that the counter didn't reset back to zero; the application\n        // is not restarted.\n        primarySwatch: Colors.blue,\n      ),\n      home: const MyHomePage(title: 'Flutter Demo Home Page'),\n    );\n  }\n}\n\nclass MyHomePage extends StatefulWidget {\n  const MyHomePage({super.key, required this.title});\n\n  // This widget is the home page of your application. It is stateful, meaning\n  // that it has a State object (defined below) that contains fields that affect\n  // how it looks.\n\n  // This class is the configuration for the state. It holds the values (in this\n  // case the title) provided by the parent (in this case the App widget) and\n  // used by the build method of the State. Fields in a Widget subclass are\n  // always marked \"final\".\n\n  final String title;\n\n  @override\n  State<MyHomePage> createState() => _MyHomePageState();\n}\n\nclass _MyHomePageState extends State<MyHomePage> {\n  int _counter = 0;\n\n  void _incrementCounter() {\n    setState(() {\n      // This call to setState tells the Flutter framework that something has\n      // changed in this State, which causes it to rerun the build method below\n      // so that the display can reflect the updated values. If we changed\n      // _counter without calling setState(), then the build method would not be\n      // called again, and so nothing would appear to happen.\n      _counter++;\n    });\n  }\n\n  @override\n  Widget build(BuildContext context) {\n    // This method is rerun every time setState is called, for instance as done\n    // by the _incrementCounter method above.\n    //\n    // The Flutter framework has been optimized to make rerunning build methods\n    // fast, so that you can just rebuild anything that needs updating rather\n    // than having to individually change instances of widgets.\n    return Scaffold(\n      appBar: AppBar(\n        // Here we take the value from the MyHomePage object that was created by\n        // the App.build method, and use it to set our appbar title.\n        title: Text(widget.title),\n      ),\n      body: Center(\n        // Center is a layout widget. It takes a single child and positions it\n        // in the middle of the parent.\n        child: Column(\n          // Column is also a layout widget. It takes a list of children and\n          // arranges them vertically. By default, it sizes itself to fit its\n          // children horizontally, and tries to be as tall as its parent.\n          //\n          // Invoke \"debug painting\" (press \"p\" in the console, choose the\n          // \"Toggle Debug Paint\" action from the Flutter Inspector in Android\n          // Studio, or the \"Toggle Debug Paint\" command in Visual Studio Code)\n          // to see the wireframe for each widget.\n          //\n          // Column has various properties to control how it sizes itself and\n          // how it positions its children. Here we use mainAxisAlignment to\n          // center the children vertically; the main axis here is the vertical\n          // axis because Columns are vertical (the cross axis would be\n          // horizontal).\n          mainAxisAlignment: MainAxisAlignment.center,\n          children: <Widget>[\n            const Text(\n              'You have pushed the button this many times:',\n            ),\n            Text(\n              '$_counter',\n              style: Theme.of(context).textTheme.headline4,\n            ),\n          ],\n        ),\n      ),\n      floatingActionButton: FloatingActionButton(\n        onPressed: _incrementCounter,\n        tooltip: 'Increment',\n        child: const Icon(Icons.add),\n      ), // This trailing comma makes auto-formatting nicer for build methods.\n    );\n  }\n}\n","tokenPosTable":[[12,1337,29],[13,1338,5,1339,12],[14,1340,14],[15,1341,14],[25,1342,31],[29,1343,3],[8,1344,22,1345,9,1346,22,1347,25,1348,27],[44,1349,16],[47,1350,38,1351,56],[33,1352,27,1353,46,1354,9,1355,27,1356,30,1357,53],[61,1358,7],[54,1359,5,1360,14],[60,1361,17,1362,7,1363,7,1364,15,1365,15],[62,1366,3],[65,1367,29],[72,1368,5,1369,12],[73,1370,15],[76,1371,16,1372,21,1373,28],[78,1374,13],[81,1375,16],[96,1376,48],[97,1377,29],[101,1378,13],[102,1379,26,1380,17],[103,1381,28,1382,40,1383,50,1384,31,1385,38],[108,1386,29],[109,1387,20],[110,1388,18],[114,1389,3],[51,1390,7,1391,18],[115,1392,1],[4,1393,24,1394,3],[5,1395,1]]},"id":"15"}
```

However when using the Web Server device and Dart Debug extension, it fails:

```
[12:35:06 PM] [VmService] [Info] [Flutter (Web Server)] ==> {"id":"15","jsonrpc":"2.0","method":"getObject","params":{"isolateId":"1","objectId":"4"}}
[12:35:06 PM] [VmService] [Info] [Flutter (Web Server)] <== {"jsonrpc":"2.0","error":{"code":-32603,"message":"getObject: Invalid argument(s): Uri http://localhost:50403/packages/dartcode_flutter_4208/main.dart must have scheme 'file:'.","data":{"details":"#0      PosixStyle.pathFromUri (package:path/src/style/posix.dart:55:5)\n#1      Context.fromUri (package:path/src/context.dart:1000:32)\n#2      FileSystem.getPath (package:file/src/interface/file_system.dart:158:24)\n#3      LocalFileSystem.file (package:file/src/backends/local/local_file_system.dart:26:54)\n#4      ErrorHandlingFileSystem.file (package:flutter_tools/src/base/error_handling_io.dart:123:24)\n#5      WebAssetServer._resolveDartFile (package:flutter_tools/src/isolated/devfs_web.dart:537:20)\n#6      WebAssetServer.dartSourceContents (package:flutter_tools/src/isolated/devfs_web.dart:590:25)\n#7      AppInspector._getScript (package:dwds/src/debugging/inspector.dart:404:39)\n#8      AppInspector.getObject (package:dwds/src/debugging/inspector.dart:386:22)\n<asynchronous suspension>\n#9      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)\n<asynchronous suspension>\n","request":{"id":"15","jsonrpc":"2.0","method":"getObject","params":{"isolateId":"1","objectId":"4"}}}},"id":"15"}
```

```
getObject: Invalid argument(s): Uri http://localhost:50403/packages/dartcode_flutter_4208/main.dart must have scheme 'file:'

#0      PosixStyle.pathFromUri (package:path/src/style/posix.dart:55:5)
#1      Context.fromUri (package:path/src/context.dart:1000:32)
#2      FileSystem.getPath (package:file/src/interface/file_system.dart:158:24)
#3      LocalFileSystem.file (package:file/src/backends/local/local_file_system.dart:26:54)
#4      ErrorHandlingFileSystem.file (package:flutter_tools/src/base/error_handling_io.dart:123:24)
#5      WebAssetServer._resolveDartFile (package:flutter_tools/src/isolated/devfs_web.dart:537:20)
#6      WebAssetServer.dartSourceContents (package:flutter_tools/src/isolated/devfs_web.dart:590:25)
#7      AppInspector._getScript (package:dwds/src/debugging/inspector.dart:404:39)
#8      AppInspector.getObject (package:dwds/src/debugging/inspector.dart:386:22)
<asynchronous suspension>
#9      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:125:47)
<asynchronous suspension>
```

I suspect this is a DWDS issue, as we're not passing a URI here, we're just calling `getScript`. I'll verify it still occurs on latest code and if so we can move the issue there.

I've also opened https://github.com/Dart-Code/Dart-Code/issues/4209 because it looks like the step that's failing (fetching a script so we can map `tokenPos` to `line`/`col`) might not even be necessary anymore (it's not provided in the initial source locations).

**Comment by DanTup** [comment]
Looks like this has already been fixed in DWDS/Flutter, as it's working fine on the `master` branch:

<img width="1420" alt="Screenshot 2022-10-07 at 12 56 06" src="https://user-images.githubusercontent.com/1078012/194547607-56c6f979-468f-47ba-81b8-19475961a707.png">

Unfortunately I can't think of a workaround, but I'll take a look at #4209 early next week because if that side-steps the issue, I can push it in a pre-release build of the Dart extension that could get you a fix for stable in the meantime.

**Comment by DanTup** [comment]
(Closing this one as the real issue is fixed in Flutter/DWDS, we can follow up on #4209 about whether we can avoid calling the bad code)

**Comment by maks** [comment]
Thanks for looking into this so quickly @DanTup 👍
As I only need to debug our web build occasionally,I'm fine to work around this for now but switching to Flutter master when needed until the fix makes its way down to stable.
